src/Components/Web/src/Routing/NavigationLock.cs

```diff
@@ -38,26 +38,45 @@ public sealed class NavigationLock : IComponent, IAsyncDisposable
 
     void IComponent.Attach(RenderHandle renderHandle)
     {
+        _renderHandle = renderHandle;
     }
 
-    async Task IComponent.SetParametersAsync(ParameterView parameters)
+    Task IComponent.SetParametersAsync(ParameterView parameters)
     {
-        var lastHasOnBeforeInternalNavigationCallback = HasOnBeforeInternalNavigationCallback;
-        var lastConfirmExternalNavigation = ConfirmExternalNavigation;
+        foreach (var parameter in parameters)
+        {
+            if (parameter.Name.Equals(nameof(OnBeforeInternalNavigation), StringComparison.OrdinalIgnoreCase))
+            {
+                OnBeforeInternalNavigation = (EventCallback<LocationChangingContext>)parameter.Value;
+            }
+            else if (parameter.Name.Equals(nameof(ConfirmExternalNavigation), StringComparison.OrdinalIgnoreCase))
+            {
+                ConfirmExternalNavigation = (bool)parameter.Value;
+            }
+            else
+            {
+                throw new ArgumentException($"The component '{nameof(NavigationLock)}' does not accept a parameter with the name '{parameter.Name}'.");
+            }
+        }
 
-        parameters.SetParameterProperties(this);
```

Comment:
I guess I'm missing something basic here, but why replace `SetParameterProperties` with the logic above? Isn't it equivalent?

Reply:
It is equivalent, but I noticed that other types implementing `IComponent` directly had similar logic, presumably for perf reasons. But I'm happy to change it back if you think it's worth reducing the size of the changes in this PR.

Reply:
OK that's fine - I just wanted to understand and was unsure if this reflected some behavioral change. LGTM.